provider/image.go

```diff
@@ -212,7 +218,12 @@ func (p *dockerNativeProvider) dockerBuild(ctx context.Context,
 		}
 
 		// we use the credentials for the server declared in the program, looking them up from the host authConfigs.
```

Comment:
this comment should probably include the gist of your comment on L226, and say "looking up from `cfg` rather that `authConfigs`.